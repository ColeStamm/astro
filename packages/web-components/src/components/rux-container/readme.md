# rux-container



<!-- Auto Generated Below -->


## Slots

| Slot          | Description     |
| ------------- | --------------- |
| `"(default)"` | The input label |


## Shadow Parts

| Part        | Description                        |
| ----------- | ---------------------------------- |
| `"body"`    | The container of the suffix slot   |
| `"footer"`  |                                    |
| `"header"`  | The asterisk when required is true |
| `"tab-bar"` | The container of the prefix slot   |
| `"toolbar"` | The container of the suffix slot   |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
