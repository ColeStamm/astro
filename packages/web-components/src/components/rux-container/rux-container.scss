:host {
    display: block;
    --container-header-padding: 16px 20px;
    --container-footer-padding: 20px;
    --container-header-background-color: var(--color-backgrund-surface-header);
}

.hidden,
:host([hidden]) {
    display: none;
}

.rux-container {
    background: var(--color-background-surface-default);
}

.rux-container__header {
    padding: var(--container-header-padding);
    background: var(--color-background-surface-header);
    font-family: var(--font-heading-2-font-family);
    font-size: var(--font-heading-2-font-size);
    font-weight: var(--font-heading-2-font-weight);
    letter-spacing: var(--font-heading-2-letter-spacing);
}

.rux-container__tab-bar {
    background: var(--color-background-surface-header);
    padding: 0 10px;
}

.rux-container__toolbar {
    background: var(--color-background-surface-header);
    padding: var(--container-header-padding);
}

.rux-container__footer {
    background: var(--color-background-surface-default);
    padding: var(--container-footer-padding);
}
