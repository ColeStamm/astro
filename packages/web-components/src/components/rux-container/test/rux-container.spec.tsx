import { newSpecPage } from '@stencil/core/testing'
import { RuxContainer } from '../rux-container'

describe('rux-container', () => {
    it('renders', async () => {
        const page = await newSpecPage({
            components: [RuxContainer],
            html: `<rux-container></rux-container>`,
        })
        expect(page.root).toMatchSnapshot()
    })
})
