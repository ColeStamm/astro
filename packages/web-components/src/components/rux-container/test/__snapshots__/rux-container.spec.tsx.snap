// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rux-container renders 1`] = `
<rux-container>
  <mock:shadow-root>
    <div class="rux-container">
      <div class="hidden rux-container__header" part="header">
        <slot name="header"></slot>
      </div>
      <div class="hidden rux-container__tab-bar" part="tab-bar">
        <slot name="tab-bar"></slot>
      </div>
      <div class="hidden rux-container__toolbar" part="toolbar">
        <slot name="toolbar"></slot>
      </div>
      <div class="rux-container__body" part="body">
        <slot></slot>
      </div>
      <div class="hidden rux-container__footer" part="footer">
        <slot name="footer"></slot>
      </div>
    </div>
  </mock:shadow-root>
</rux-container>
`;
